server/test/logger.test.ts

```diff
@@ -0,0 +1,46 @@
+import { logger, logStackTrace } from "../src/logger";
+import { transports } from "winston";
+import { Writable } from "stream";
+
+let output = "";
+const stream = new Writable();
+stream._write = (chunk, encoding, next) => {
+  output = output += chunk.toString();
+  next();
+};
+const streamTransport = new transports.Stream({ stream });
+logger.add(streamTransport);
```

Comment:
Quick last minute thought — you could clean up the test output by removing the current transports:

```suggestion
for (const transport of logger.transports.slice()) {
  logger.remove(transport);
}
logger.add(streamTransport);
```